pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -392,15 +392,15 @@ public void sendComplete(Exception exception) {
             if (exception != null) {
                 log.error("[{}][{} -> {}] Error producing on remote broker", replicator.topicName,
                         replicator.localCluster, replicator.remoteCluster, exception);
-                return;
-            }
-
-            if (log.isDebugEnabled()) {
-                log.debug("[{}][{} -> {}] Message persisted on remote broker", replicator.topicName,
-                        replicator.localCluster, replicator.remoteCluster);
+                // cursor shoud be rewinded since it was incremented when readMoreEntries
+                replicator.cursor.rewind();
```

Comment:
Rewind is a _cheap_ operation, it just resets the `readPosition` to `markDeletePosition + 1`